include/multipass/platform.h

```diff
@@ -36,13 +36,22 @@ namespace multipass
 {
 namespace platform
 {
+std::map<QString, QString> extra_settings_defaults();
+
+QString interpret_setting(const QString& key, const QString& val);
+void sync_winterm_profiles();
```

Comment:
This should also be `client_final_setup` or so?

Reply:
I think it is preferable to be more specific about the operations being provided at this level. That leaves it to the CLI client (or any other calling code) to decide when to call. And it increases modularity, allowing client code to call this bit without that bit, and it is more coherent with the current granularity: we have things like `make_logger`, `make_update_prompt`, `setup_gui_autostart_prerequisites`, rather than just `client_setup`. The platform code is concerned with the implementation of specific operations in different platforms, the client code is concerned with organizing those operations.

Reply:
Yeah, but suddenly you have winterm-specific code in all platforms, even if it's just NOOP.

If you look at https://github.com/canonical/multipass/issues/756#issuecomment-621127639 - our longer-term plan for "integrations" - to follow what you're doing, you'd need an explicit call to each of the enabled integrations here.

IMO the client should have entry points for the integrations (pre_, post_setup, modify_request), but it should / need not have the context to decide whether to run them or not.

Reply:
So you envision more "syncs" needed for other app integrations? I am not sure what the best approach will be for them, but I would prefer to decide separately. Creating general entry points here means assuming, in this PR, future syncs at specific points. Instead, we could generalize this then, as needed, when designing/implementing those features.

Also, the way I see it, things like pre/post make more sense at the client level. The implication of chronology or precedence is meaningful in the CLI as relative to a sequence of tasks being carried out there. Less so in the GUI (we could think of them as relative to the main loop, but that would be too frequent a period). So I guess that'd be something to improve when implementing the WT sync on the GUI side. At the platform level, however, there is no inherent sequence to speak of, so introducing it there feels like kind of polluting it somewhat.

Reply:
How about:

1) In this PR, we rename this as `app_integration_sync` and only call the winterm sync on windows, (separate function residing in the unnamed namespace)
2) In future PRs, we refine as needed

:grey_question:

Reply:
(not sure you get notified about the :+1: above, so…)

:+1:

Reply:
Hehe, cool! :ok_hand:

Reply:
And yes, I was only notified of the comment.

Reply:
For (my own) future reference: this would pose problems with testing, unless we kept the granularity at a separate level. After discussing on IRC, we decided such refactoring now would probably be thrown away soon and not worth it. In the future, we should avoid providing NOOP syncs at the platform level, for instance, with some plugin structure providing hooks for different sync "moments".